WPMedia/engine-theme-sdk
# tmedia 653 Use action for caching uniformly across actions in use

## Description
Ensure github action caching uniformity 

## Jira Ticket
- [TMEDIA-653](https://arcpublishing.atlassian.net/jira/software/c/projects/TMEDIA/boards/875?modal=detail&selectedIssue=TMEDIA-653)

## Acceptance Criteria
Github actions caching is uniformly done throughout blocks, themes components, and engine theme sdk

Write a doc in the themes internal space around caching and best practices there 

Make sure that the cache is reset when there’s a change to the package-lock and used when there’s no change to the package-lock

## Test Steps
- See example in lerna playground where this structure and investigation works https://github.com/WPMedia/themes-lerna-playground/pull/15

## Effect Of Changes
### Before
- Unnecessary caching and potentially not enough installing

### After
- Unnecessary caching and potentially not enough installing
- all pr actions pass as expected 

## Dependencies or Side Effects

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).
